internal/rules.go

```diff
@@ -315,5 +317,12 @@ func interpolateTags(pattern string, tagSet api.TagSet) (string, error) {
 			buffer.WriteString(token)
 		}
 	}
+
+	for key := range tagSet {
+		if !usedTags[key] && enforceAllTagsUsed {
+			return "", errors.New("unused key " + key + " in tagset")
```

Comment:
this is more of a nit, but let's ensure a subtype of `ConversionError` is used. See `newMissingTag` in `errors.go`